dask/dataframe/core.py

```diff
@@ -8034,31 +8035,51 @@ def maybe_shift_divisions(df, periods, freq):
 
 @wraps(pd.to_datetime)
 def to_datetime(arg, meta=None, **kwargs):
-    tz_kwarg = {"tz": "utc"} if kwargs.get("utc") else {}
+    if PANDAS_GT_200 and "infer_datetime_format" in kwargs:
+        warnings.warn(
+            "The argument 'infer_datetime_format' is deprecated and will be removed in a future version. "
+            "A strict version of it is now the default, see "
+            "https://pandas.pydata.org/pdeps/0004-consistent-to-datetime-parsing.html. "
+            "You can safely remove this argument.",
+            UserWarning,
+        )
+        kwargs.pop("infer_datetime_format")
+
     if meta is None:
         if isinstance(arg, Index):
-            meta = pd.DatetimeIndex([], **tz_kwarg)
+            meta = get_meta_library(arg).DatetimeIndex(
+                [], **{"tz": "utc"} if kwargs.get("utc") else {}
+            )
             meta.name = arg.name
         elif not (is_dataframe_like(arg) or is_series_like(arg)):
             raise NotImplementedError(
                 "dask.dataframe.to_datetime does not support "
                 "non-index-able arguments (like scalars)"
             )
         else:
-            meta = meta_series_constructor(arg)([pd.Timestamp("2000", **tz_kwarg)])
+            if not isinstance(arg, _Frame):
+                arg_meta = arg.head(2)
+            else:
+                if is_series_like(arg):
+                    arg_meta = (
+                        meta_series_constructor(arg)(["2000/1/1"])
+                        if arg.dtype.kind == "O"
+                        else arg._meta_nonempty
+                    )
+                else:
+                    arg_meta = arg._meta_nonempty
+                    assign_dict = dict()
+                    for col in arg_meta.columns:
+                        if arg_meta[col].dtype.kind == "O":
+                            assign_dict[col] = "2000" if col == "year" else "1"
+                        elif col == "year" and is_numeric_dtype(arg_meta[col]):
+                            assign_dict[col] = 2000
+                    arg_meta = arg_meta.assign(**assign_dict)
+            meta = get_meta_library(arg).to_datetime(arg_meta, **kwargs)
```

Comment:
Previously we were passing `tz_kwarg` here. Is that no longer needed?

Reply:
From what I understand, `tz_kwarg` was used here to grab only the relevant `to_datetime` kwarg for the Series constructor - now that we're constructing the meta directly with `to_datetime`, we should be good to pass all of these kwargs directly in (with the exception of `infer_datetime_format` for pandas 2)